bin/test.py

```diff
@@ -40,24 +47,24 @@ def get_current_hash():
     return get_output('git rev-parse HEAD')
 
 
-def build_saas(sm_hash: str):
-    clone_dir = '/tmp' / pathlib.Path('saas')
-    web_dir = 'web'
-    run(f'rm -rf {clone_dir}; git clone {SAAS_REPO} {clone_dir} --recurse-submodules')
-    run(f'cd {clone_dir} && git checkout master && git pull')
-    run(f'cd {clone_dir}/{web_dir}/{SM_DIR} && git checkout {sm_hash}')
-    run(f'cd {clone_dir} && make -C {web_dir} node_modules/done.stamp build')
+def setup_user(user, sm_hash: str):
+    clone_dir = '/tmp' / pathlib.Path(user['name'])
+    web_dir = clone_dir / user['web_dir']
+    run(f'rm -rf {clone_dir}; git clone {user["repo"]} {clone_dir} --recurse-submodules')
+    run(f'git checkout master && git pull', cwd=clone_dir)
+    run(f'git checkout {sm_hash}', cwd=web_dir/SM_DIR)
 
 
-def build_core(sm_hash: str):
-    clone_dir = '/tmp' / pathlib.Path('core')
-    web_dir = 'webui/react'
-    run(f'rm -rf {clone_dir}; git clone {CORE_REPO} {clone_dir} --recurse-submodules')
-    run(f'cd {clone_dir} && git checkout master && git pull')
-    run(f'cd {clone_dir}/{web_dir}/{SM_DIR} && git checkout {sm_hash}')
-    run(f'cd {clone_dir} && make -C {web_dir} node_modules/done.stamp build')
+def test(sm_hash: str):
+    for user in users:
+        setup_user(user, sm_hash)
+        clone_dir = '/tmp' / pathlib.Path(user['name'])
+        web_dir = clone_dir / user['web_dir']
+        for target in ['get-deps', 'check', 'build', 'test']:
```

Comment:
maybe better off modifying the respective makefiles to have a target that does all these?

Reply:
hmm we could do that. I think I'd probably still use the specific separate targets so that the tester has control over the steps that are being run. this tester would mostly be deprecated once we get circle ci to run these steps so that we can benefit from their ui